src/databricks/labs/ucx/contexts/application.py

```diff
@@ -229,6 +230,51 @@ def tables_migrator(self):
             self.principal_acl,
         )
 
+    @cached_property
+    def table_migrator_hiveserde_parquet(self):
+        return TablesMigrator(
+            self.tables_crawler,
+            self.grants_crawler,
+            self.workspace_client,
+            self.sql_backend,
+            self.table_mapping,
+            self.group_manager,
+            self.migration_status_refresher,
+            self.principal_acl,
+            hiveserde_in_place_migrate=HiveSerdeType.PARQUET,
```

Comment:
There has to be only one instance of TableMigrator in GlobalContext. If you have three different strategies of in-place migration, then they have to be dedicated methods. Or classes, if they're big

Reply:
@nfx Please have a look on the new commit. 
The original idea of `hiveserde_in_place_migrate` is that the migration of PARQUET, ORC and AVRO serde can be done in individual tasks, but in the new commit I consolidate them to make it simpler so one task will migrate them all together.
But now we still have two `TableMigrator` in `GlobalContext`, the `table_migrator_hiveserde_in_place` that has `hiveserde_in_place_migrate=True` will be used only by the hiveserde in place migrate workflow, and the `tables_migrator` that has `hiveserde_in_place_migrate=False` will be used by all other table migration workflows and also the CTAS based hiveserde migration. Eventually user will decide if they want run hiveserde in place or CTAS migration workflow.